### DISCUSSION TITLE
Running RKE2 completely on an NFS server

### DISCUSSION THREAD

**Discussion by davidgaster** [rancher/rke2#6600] [General]
Hi, after researching a few other github discussions, the consensus is that operating directly on NFS is not recommended, and a compatible filesystem like ext4 is preferred. However, our embedded system mandates that the entire container runtime is mounted to an NFS server on boot. I'm exploring `fuse-overlayfs` as an alternative to the default `overlayfs` and `native` containerd snapshotters, as I've hit a roadblock due to lack of extended attribute support in Linux kernels before 5.11, and our NFS server can't be updated to a newer kernel.

Before proceeding with trying to get `fuse-overlayfs` working, I'd like to confirm whether this approach is even viable with these constraints. Any advice or suggestions would be greatly appreciated!

**Comment by brandond** [comment]
It might work, but I don't think it would work well. Why would you want to put your container runtime on a shared filesystem? They are not in any way designed to be shared between nodes.

**Reply by davidgaster** [reply]
Yeah it's not a matter of wanting to, it's that we have to for this particular use case. We're exploring other options like rootless podman as these will all be single node deployments. The attractive thing about RKE2 are the security requirements it inherently meets via `cis` (we currently leverage it for other traditional data center use cases).
